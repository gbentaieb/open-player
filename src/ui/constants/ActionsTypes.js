// Config Actions
export const MERGE_CONFIG = 'MERGE_CONFIG';

// Core Actions
export const SET_PLAYER_STATE = 'SET_PLAYER_STATE';
export const REQUEST_PLAY = 'REQUEST_PLAY';
export const SEEK_TO = 'SEEK_TO';
export const SET_PLAYER_TIMES = 'SET_PLAYER_TIMES';
