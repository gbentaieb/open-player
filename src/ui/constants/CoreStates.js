export const PLAYING = 'PLAYING';
export const SEEKING = 'SEEKING';
export const BUFFERING = 'BUFFERING';
export const STOPPED = 'STOPPED';
export const LOADING = 'LOADING';
export const LOADED = 'LOADED';
export const ENDED = 'ENDED';
export const PAUSED = 'PAUSED';

export const CORE_STATES = [PLAYING, SEEKING, BUFFERING, STOPPED, LOADING, LOADED, ENDED, ENDED];
