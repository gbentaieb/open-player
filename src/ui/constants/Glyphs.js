import playGlyph from '../assets/icons/play.svg';
import pauseGlyph from '../assets/icons/pause.svg';

export const PLAY_GLYPH = playGlyph;
export const PAUSE_GLYPH = pauseGlyph;
