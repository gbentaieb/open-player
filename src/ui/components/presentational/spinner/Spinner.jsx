import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { BUFFERING, LOADING, LOADED, SEEKING } from '../../../constants/CoreStates';

export const isWaiting = playerState => (
  [BUFFERING, LOADING, LOADED, SEEKING].includes(playerState)
);

const styles = theme => ({
  progress: {
    margin: theme.spacing.unit * 2,
  },
});

const Spinner = props => (
  isWaiting(props.playerState) &&
    <div>
      <CircularProgress className={props.classes.progress} />
    </div>
);

Spinner.propTypes = {
  classes: PropTypes.object.isRequired,
  playerState: PropTypes.string.isRequired,
};

export { Spinner };
export default withStyles(styles)(Spinner);
