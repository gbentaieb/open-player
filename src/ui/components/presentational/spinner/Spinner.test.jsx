import React from 'react';
import { Spinner, isWaiting } from './Spinner';
import { BUFFERING, LOADING, LOADED, SEEKING, PLAYING } from '../../../constants/CoreStates';

const defaultProps = {
  classes: {},
  playerState: PLAYING,
};

describe('Spinner', () => {
  test('Spinner > Snapshot', () => {
    const wrapper = shallow(<Spinner {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Spinner > isWaiting', () => {
    [BUFFERING, LOADING, LOADED, SEEKING].forEach((state) => {
      expect(isWaiting(state)).toBe(true);
    });

    expect(isWaiting(PLAYING)).toBe(false);
  });
});
