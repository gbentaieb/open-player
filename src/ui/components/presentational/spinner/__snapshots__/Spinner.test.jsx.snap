// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Spinner > Snapshot 1`] = `""`;
